server/src/main/java/org/opensearch/cluster/routing/RoutingNode.java

```diff
@@ -329,6 +330,21 @@ public List<ShardRouting> shardsWithState(ShardRoutingState... states) {
         return shards;
     }
 
+    /**
+     * Determine the shards that satisfy the predicate
+     * @param predicate shard routing predicate which needs to be met
+     * @return List of shards
+     */
+    public List<ShardRouting> shardsSatisfyingPredicate(Predicate<ShardRouting> predicate) {
```

Comment:
We could probably move this method to `DiskThresholdDecider` (package private or private), since this is the only place it is used

Reply:
I can restrict it but we might need it going forward with other remote storage work which is coming up soon.
Do we move it to here later?

Reply:
Sure